engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -799,6 +807,10 @@ public void monitorHeartbeat()
     processEvents();
 
     committedWindowId = updateCheckpoints(false);
+    if (lastCommittedWindowId != committedWindowId) {
```

Comment:
There is a merge conflict here. Also is this being added so that multiple calls don't happen for the same committed window id.